main_test.go

```diff
@@ -44,51 +89,230 @@ webhook:
+					},
+					Webhook: ctrlconfigv1alpha1.ControllerWebhook{
+						Port: pointer.Int(defaultWebhookPort),
+					},
+				},
+				InternalCertManagement: enableDefaultInternalCertManagement,
+			},
+			wantOptions: defaultControlOptions,
 		},
 		{
-			name:                             "port overwrite config",
-			configFile:                       portOverWriteConfig,
-			wantPort:                         9444,
-			wantEnableInternalCertManagement: true,
+			name:       "ControllerManagerConfigurationSpec overwrite config",
+			configFile: ctrlManagerConfigSpecOverWriteConfig,
+			wantConfiguration: configv1alpha2.Configuration{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: configv1alpha2.GroupVersion.String(),
+					Kind:       "Configuration",
+				},
+				Namespace:                  pointer.String(defaultNamespace),
+				ManageJobsWithoutQueueName: false,
+				ControllerManagerConfigurationSpec: cfg.ControllerManagerConfigurationSpec{
+					Health: ctrlconfigv1alpha1.ControllerHealth{
+						HealthProbeBindAddress: ":38081",
+					},
+					Metrics: ctrlconfigv1alpha1.ControllerMetrics{
+						BindAddress: ":38080",
+					},
+					LeaderElection: &componentconfigv1alpha1.LeaderElectionConfiguration{
+						LeaderElect:  pointer.Bool(true),
+						ResourceName: "test-id",
+					},
+					Webhook: ctrlconfigv1alpha1.ControllerWebhook{
+						Port: pointer.Int(9444),
+					},
+				},
+				InternalCertManagement: enableDefaultInternalCertManagement,
+			},
+			wantOptions: ctrl.Options{
+				HealthProbeBindAddress: ":38081",
+				MetricsBindAddress:     ":38080",
+				Port:                   9444,
+				LeaderElection:         true,
+				LeaderElectionID:       "test-id",
+			},
 		},
 		{
-			name:                             "cert overwrite config",
-			configFile:                       certOverWriteConfig,
-			wantPort:                         9443,
-			wantEnableInternalCertManagement: false,
+			name:       "cert options overwrite config",
+			configFile: certOverWriteConfig,
+			wantConfiguration: configv1alpha2.Configuration{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: configv1alpha2.GroupVersion.String(),
+					Kind:       "Configuration",
+				},
+				Namespace:                  pointer.String(defaultNamespace),
+				ManageJobsWithoutQueueName: false,
+				ControllerManagerConfigurationSpec: cfg.ControllerManagerConfigurationSpec{
+					Health: ctrlconfigv1alpha1.ControllerHealth{
+						HealthProbeBindAddress: defaultHealthProbeAddress,
+					},
+					Metrics: ctrlconfigv1alpha1.ControllerMetrics{
+						BindAddress: defaultMetricsAddress,
+					},
+					LeaderElection: &componentconfigv1alpha1.LeaderElectionConfiguration{
+						LeaderElect:  pointer.Bool(true),
+						ResourceName: defaultLeaderElectionID,
+					},
+					Webhook: ctrlconfigv1alpha1.ControllerWebhook{
+						Port: pointer.Int(defaultWebhookPort),
+					},
+				},
+				InternalCertManagement: enableDefaultInternalCertManagement,
```

Comment:
You're correct.
As you say in https://github.com/kubernetes-sigs/kueue/pull/362#discussion_r962088288, I missed this bug because I entered the wrong test data.

> It should be webhookServiceName then. Also secretName.

Thanks!

Reply:
Done.